src/chatty/util/history/HistoryManager.java

```diff
@@ -0,0 +1,206 @@
+package chatty.util.history;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+import chatty.Room;
+import chatty.User;
+import chatty.util.UrlRequest;
+import chatty.util.UrlRequest.FullResult;
+
+import chatty.util.irc.MsgTags;
+import chatty.util.settings.Settings;
+
+
+/**
+ * History Manager which should be the entry point for getting historic Chat messages from external services.
+ * Currently, only robotty https://recent-messages.robotty.de is implemented
+ * @author m00hlti
+ */
+public class HistoryManager {
+    private enum MessageType {
+        UNDEFINED,
+        PRIVATMESSAGE,
+        CLEARCHAT
+    }
+
+    private final Settings settings;
+
+    private final static String strHistoryURL = "https://recent-messages.robotty.de/api/v2/recent-messages/";
+    private final int nDefaultLimitMessages = 30;
+    private long lTimeStampBefore = 0;
```

Comment:
What are these for?

Reply:
- the strHistoryURL (now HISTORYURL) is the base url for the API call
- the nDefaultLimitMessages (now defaultLimitMessages) is the default number of messages which will be requested without any additional number. 
- lTimeStampBefore (and lTimeStampAfter) are parameters to limit the amount of messages. Totally open for discussions here - Currently limiting to 24h.